### PR TITLE
Add valid columns check in _remove_unused_columns method

### PR DESCRIPTION
**Pull request by arthasking123**
# What does this PR do?

Add valid columns check in _remove_unused_columns method to fix
https://github.com/huggingface/datasets/issues/6973
https://github.com/huggingface/datasets/issues/6535
This improvement helps users to quickly identify and resolve issues related to column matching between the dataset and the model's forward method.

## Before submitting
- [X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [X] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by arthasking123** [comment]
@muellerzr @SunMarc

File: src/transformers/models/mixtral/modeling_mixtral.py [hunk 1]
```diff
@@ -147,9 +147,10 @@ def load_balancing_loss_func(
             router_per_expert_attention_mask, dim=0
         )
 
-    overall_loss = torch.sum(tokens_per_expert * router_prob_per_expert.unsqueeze(0))
-    return overall_loss * num_experts
+    # Ensure the loss calculation balances each expert's load correctly
+    overall_loss = torch.sum(tokens_per_expert * router_prob_per_expert)
```
Comment (muellerzr) [inline src/transformers/models/mixtral/modeling_mixtral.py:151]: These seem like unrelated changes, no?

**Review by muellerzr — COMMENTED** [review]
Thanks, overall this feels like a good guard. General comment as I'm unsure if we have unrelated changes in this PR

File: src/transformers/models/mixtral/modeling_mixtral.py (hunk 1 above)
Comment (arthasking123) [reply src/transformers/models/mixtral/modeling_mixtral.py:151]: Oh ,sorry for that

**Comment by arthasking123** [comment]
@muellerzr thanks!

**Review by SunMarc — APPROVED** [review]
LGTM !

**Comment by arthasking123** [comment]
@amyeroberts would you please merge this PR? Thanks

**Review by amyeroberts — APPROVED** [review]
Thanks!
